src/packages/core/src/decorators/read-only-property.ts

```diff
@@ -1,15 +1,31 @@
 import 'reflect-metadata';
 
-const readOnlyPropertyKey = Symbol('BaseResolverReadOnlyProperty');
+const readOnlyPropertyBackendKey = Symbol('BaseResolverReadOnlyBackend');
+const readOnlyPropertyAdminUIKey = Symbol('BaseResolverReadOnlyAdminUI');
 
-export function ReadOnlyProperty() {
+type Props = {
+	backend?: boolean;
+	adminUI?: boolean;
+};
+
+export function ReadOnlyProperty({ backend, adminUI }: Props = { backend: true, adminUI: true }) {
 	return (target: any, propertyKey: string | symbol) => {
-		// Normally we'd just return Reflect.metadata(), but TypeGraphQL works on the constructor
-		// as the target, so we need to as well for later.
-		Reflect.metadata(readOnlyPropertyKey, true)(target.constructor, propertyKey);
+		if (propertyKey) {
+			if (adminUI)
+				Reflect.metadata(readOnlyPropertyAdminUIKey, true)(target.constructor, propertyKey);
+			if (backend)
+				Reflect.metadata(readOnlyPropertyBackendKey, true)(target.constructor, propertyKey);
+			return;
+		}
+		if (adminUI) Reflect.metadata(readOnlyPropertyAdminUIKey, true)(target.constructor);
```

Comment:
I think we should error if we get here as we do not want people decorating a class with this decorator?

Reply:
Will delete

Reply:
will do